lib/command.js

```diff
@@ -6,9 +9,13 @@ module.exports = function (yargs, usage, validation) {
 
   var handlers = {}
   self.addHandler = function (cmd, description, builder, handler) {
-    // allow a module to define all properties
-    if (typeof cmd === 'object' && typeof cmd.command === 'string' && cmd.builder && typeof cmd.handler === 'function') {
-      self.addHandler(cmd.command, extractDesc(cmd), cmd.builder, cmd.handler)
+    // allow modules that define (a) all properties or (b) only command and description
+    if (typeof cmd === 'object' && typeof cmd.command === 'string') {
+      if (cmd.builder && typeof cmd.handler === 'function') {
+        self.addHandler(cmd.command, extractDesc(cmd), cmd.builder, cmd.handler)
+      } else {
+        self.addHandler(cmd.command, extractDesc(cmd))
+      }
```

Comment:
@maxrimue Good question. This could certainly be done in a separate PR (maybe as a precursor) but is just something I ran into when testing with `require-directory`.

A couple reasons:

1. yargs currently blows up when you give it a command module/object that does not have all the expected properties of a command module - this is something that should probably be fixed in general
2. `require-directory` looks for and loads `json` files as well as `js` and `coffee` modules by default, which I have overridden to be just `js` (since a JSON "command module" cannot support a `handler` function by itself), but if folks want to define "commands" via JSON files, then they will need this change to support an option of `extensions: ['js', 'json']` (or similar)

I don't really think no. 2 is a big concern, unless folks are dynamically generating their commands as JSON (not sure why you would), but is more of a "why not?".

I view no. 1 as a bigger problem that we should address, either before or with #492. So figured I'd kill two birds with one stone here. I can easily be persuaded otherwise.